### ISSUE TITLE
Parse hexadecimal integers

### ISSUE THREAD

**Issue by wrobell** [crdoconnor/strictyaml#105]
YAML supports multiple integer representations

    https://yaml.org/type/int.html

for example `- 0x77` parses as `[119]`.

StrictYAML raises validation error
```
In [1]: import strictyaml
In [2]: strictyaml.load('- 0x77', strictyaml.Seq(strictyaml.Int()))
...
YAMLValidationError: when expecting an integer
found arbitrary text
  in "<unicode string>", line 1, column 1:
    - '0x77'
     ^ (line: 1)
```

This ticket is about parsing hexadecimal integers as they are quite common.

**Comment by crdoconnor** [comment]
Would a HexInt validator serve your purposes?

**Comment by wrobell** [comment]
For my purposes `Int() | HexInt()` would be more than enough. But I think that general expectation would be to follow YAML specification, i.e. `Int()` (or `YAMLInt()`?) should allow both forms.

**Comment by crdoconnor** [comment]
The reason this project exists is because I wanted to strip out extraneous
features from the YAML spec and pare it down to something much simpler and
less fragile.

I'm happy to accommodate this use case with Int() | HexInt() though.

**Comment by wrobell** [comment]
IMHO, the question is: is there a valid use case for schema to allow only integers with specific base in the schema?

**Comment by crdoconnor** [comment]
Potentially, yes. 0x1a is not really very readable for non technical
people. Use of hex is quite a niche thing.

**Comment by eulores** [comment]
Hello,
I run into the same issue. While transpiling a specification published as PDF to parseable YAML, I had to rewrite 0xFFFF (found verbatim in the PDF specification) to 65535.
IMO, non-technical people would appreciate if there is a one-to-one correlation between the specs and the YAML code.

Having said that, I'll be quite happy once the proposed `Int()|HexInt()` solution is implemented :-)

**Comment by eulores** [comment]
This is a dirty patch that I use in my code to let `Int()` also parse hexadecimal encoded integers, such as 0xFF.
```python
from strictyaml import Int

def validate_hexint(self, chunk):
  val = chunk.contents
  try:
    return int(val.replace("_", ""), 0)
  except:
    chunk.expecting_but_found("when expecting an integer")

Int.validate_scalar = validate_hexint
```

**Comment by jnichols0** [comment]
Are there plans for a HexInt validator to be part of strictyaml? I have implemented it myself but I'm curious if I wont have to in the future or shouldn't even right now.

**Comment by crdoconnor** [comment]
Yeah, I can absolutely do this. Thanks for reminding me.

Can you share me yours?

**Comment by jnichols0** [comment]
Yes definitely, mind if I submit a PR? I would love to "officially" contribute.

**Comment by crdoconnor** [comment]
Not at all, I'd love that.

Please bear in mind that the tooling around the project might have issues
due to some code rot.

**Comment by crdoconnor** [comment]
This is something I will try to fix in the next few days.

**Comment by jnichols0** [comment]
I am working on this here: https://github.com/jnichols0/strictyaml/tree/hexint

No hitchkey story yet, just tested so far with the repl.

**Comment by jnichols0** [comment]
Submitted a PR, made a story to test it.

**Comment by crdoconnor** [comment]
Thanks! I'll take a look a little later on.

**Comment by crdoconnor** [comment]
LGTM. I'll merge it and release a new version a bit later.

**Comment by jnichols0** [comment]
I realized HexInt, as implemented, doesn't support signed expressions. Python evaluates -0x1 to -1 but -0x1 doesn't pass validation for HexInt() | Int() which seems wrong. I'll submit a PR updating the regex if you agree.

**Comment by lalogonzaalo-rgb** [comment]
> YAML admite múltiples representaciones de números enteros.
> 
> ```
> https://yaml.org/type/int.html
> ```
> 
> por ejemplo `- 0x77`analiza como `[119]`.
> 
> StrictYAML genera un error de validación.
> 
> ```
> In [1]: import strictyaml
> In [2]: strictyaml.load('- 0x77', strictyaml.Seq(strictyaml.Int()))
> ...
> YAMLValidationError: when expecting an integer
> found arbitrary text
>   in "<unicode string>", line 1, column 1:
>     - '0x77'
>      ^ (line: 1)
> ```
> 
> Este ticket trata sobre el análisis de números enteros hexadecimales, ya que son bastante comunes.

>>>strictyaml.Int

**Comment by lalogonzaalo-rgb** [comment]
> Sí, por supuesto. Gracias.
> […](#)

### : maestro

**Comment by lalogonzaalo-rgb** [comment]
> > YAML admite múltiples representaciones de números enteros.
> > ```
> > https://yaml.org/type/int.html
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > por ejemplo `- 0x77`analiza como `[119]`.
> > StrictYAML genera un error de validación.
> > ```
> > In [1]: import strictyaml
> > In [2]: strictyaml.load('- 0x77', strictyaml.Seq(strictyaml.Int()))
> > ...
> > YAMLValidationError: when expecting an integer
> > found arbitrary text
> >   in "<unicode string>", line 1, column 1:
> >     - '0x77'
> >      ^ (line: 1)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Este ticket trata sobre el análisis de números enteros hexadecimales, ya que son bastante comunes.
> 
> > > > strictyaml.Int



> Sí, por supuesto. Gracias.
> […](#)
